api/app/src/command/webhook/fitbit.ts

```diff
@@ -0,0 +1,82 @@
+import { getSettingsOrFail } from "../settings/getSettings";
+import { createWebhookRequest } from "./webhook-request";
+import { processRequest, WebhookUserDataPayload, reportDevicesUsage } from "./webhook";
+import dayjs from "dayjs";
+import { ConnectedUser } from "../../models/connected-user";
+import { Constants } from "../../shared/constants";
+import { capture } from "../../shared/notifications";
+import { FitbitCollectionTypes } from "../../mappings/fitbit/constants";
+
+export type FitbitWebhookNotification = {
+  collectionType: FitbitCollectionTypes;
+  date: string;
+  ownerId: string;
+  ownerType: string;
+  subscriptionId: string;
+};
+
+export const processData = async (data: FitbitWebhookNotification[]) => {
+  console.log("Starting to process the webhook");
+
+  for (const update of data) {
+    const { collectionType, date, ownerId: fitbitUserId } = update;
+
+    let userCxId;
+    try {
+      const connectedUser = await getConnectedUserByTokenOrFail(
+        ProviderSource.fitbit,
+        fitbitUserId
+      );
+
+      const cxId = connectedUser.cxId;
+      userCxId = cxId ? cxId : undefined;
+
+      const fitbitData = await mapData(collectionType, connectedUser, date);
+
+      const payload = { users: [{ userId: connectedUser.id, ...fitbitData }] };
+
+      const settings = await getSettingsOrFail({ id: cxId });
+      const webhookRequest = await createWebhookRequest({ cxId, payload });
+      await processRequest(webhookRequest, settings);
+
+      reportDevicesUsage(connectedUser.cxId, [connectedUser.cxUserId]);
+    } catch (error) {
+      console.log("Fitbit webhook processing failed.", error);
+      capture.error(error, {
+        extra: { update, context: `webhook.fitbit.processData`, fitbitUserId, userCxId },
+      });
+    }
+  }
+};
+
+export const mapData = async (
+  collectionType: string,
+  connectedUser: ConnectedUser,
+  startdate: string
+): Promise<WebhookUserDataPayload> => {
+  const payload: WebhookUserDataPayload = {};
+  const provider = Constants.PROVIDER_MAP[ProviderSource.fitbit];
+
+  if (collectionType === FitbitCollectionTypes.activities) {
+    const activity = await provider.getActivityData(connectedUser, startdate);
+    payload.activity = [activity];
+  } else if (collectionType === FitbitCollectionTypes.body) {
+    const body = await provider.getBodyData(connectedUser, startdate);
+    payload.body = [body];
+  } else if (collectionType === FitbitCollectionTypes.foods) {
+    const nutrition = await provider.getNutritionData(connectedUser, startdate);
+    payload.nutrition = [nutrition];
+  } else if (collectionType === FitbitCollectionTypes.sleep) {
+    const sleep = await provider.getSleepData(connectedUser, dayjs(startdate).format("YYYY-MM-DD"));
+    payload.sleep = [sleep];
+  } else {
+    capture.error(`Unrecognized collection type: ${collectionType}`, {
+      extra: { context: "fitbit.webhook.mapData" },
+    });
```

Comment:
nit:
- `capture.error` is to be used with `Error` - the compiler doesn't complain b/c Error is (unfortunately) any, but what we want on this case is `capture.message`
- Ideally we would use a constant message/text when notifying Sentry - when we add variations it creates different issues there, which makes it harder to group related issues like the ones that would arise from this scenario - in this case, its better to add `collectionType` as a property on `extra`
- let's add `connectedUser` on `extra` as well, this helps debugging/supporting users